content/control-origin-security-context.md

```diff
@@ -0,0 +1,117 @@
+---
+title: "Control Origin Security Context"
+date: 2018-04-18T20:59:55+02:00
+---
+
+## Summary
+
+Distributed protocols require additional controls for how Origin-based security
+context is defined.  Every content root identified by means of
+cryptographically-safe hash should have its own Origin and be marked as a
+Secure Context.
+
+## Motivation
+
+Current web application security model relies on the concept of [same-origin
+policy](https://en.wikipedia.org/wiki/Same-origin_policy). 
+[RFC 6454](https://tools.ietf.org/html/rfc6454) defines the "Origin" of a URI as the triple `{protocol, host, port}`.
+Two resources are of the same Origin if all values from the triple are the same.
+
+Web pages of different Origins are isolated from each other. Guarantees
+provided by this security perimeter enable web app developers to safely store
+site-specific data (local storage, cookies) and secrets (session cookies,
+access tokens) on user's machine.
+
+**Problem:** In the distributed world a location (host and port) is no longer
+relevant. Resources are _content-addressed_ and location authority is
+superseded by cryptographically-safe Content Identifiers (CIDs).
+
+The only reliable way to work around this is to create artificial subdomains
+for hostname-safe HTTP content addressing using cryptographic hashes (eg.
+[HSHCA](https://github.com/neocities/hshca)). A subdomain provides a separate
+Origin which creates an isolated security context.  Unfortunately, this
+approach obfuscates canonical representations of content-addressing and is not
+feasible for local p2p services listening on `127.0.0.1`.
+
+Historically, same-origin policies were extended to define roughly compatible
```

Comment:
I did not know that, but I suppose they must have. Can you dig out a citation of this, I think it helps the case a lot to show that there is precedent for this.